### PR TITLE
Expand CustomRequestLog %u

### PR DESCRIPTION
**Pull request by cstamas**
Add argument to invoke `request#getRemoteUser()`.

### CODE REVIEW

**Review by joakime — DISMISSED** [review]
Seems fine to me.

**Comment by joakime** [comment]
Your commits do not satisfy the Eclipse Legal requirement for a signed ECA. (See [details of commit checks](https://accounts.eclipse.org/legal/eca/validation/20175))
We are unable to merge without a signed ECA.
See https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/CONTRIBUTING.md#eclipse-contributor-agreement

File: jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java [hunk 1]
```diff
@@ -868,8 +869,12 @@ else if (arg.equalsIgnoreCase("clf"))
                 String method;
                 if (arg == null || arg.isEmpty())
                     method = "logRequestAuthenticationWithDeferred";
-                else
+                else if ("d".equals(arg))
                     method = "logRequestAuthentication";
```
Comment (lachlan-roberts) [inline jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java:873]: The logic for the pre-existing code seems incorrect.
This should be something like:
```java
if (StringUtil.isEmpty(arg))
    method = "logRequestAuthentication";
else if ("d".equals(arg))
    method = "logRequestAuthenticationWithDeferred";
else
    throw new IllegalArgumentException("Invalid arg for %u: " + arg);
```

**Review by lachlan-roberts — CHANGES_REQUESTED** [review]
Thanks for your work on this, but it looks to me like `%{d}u` and `%{r}u` are semantically the same thing. So I don't think there is a need for both.

You can follow the implementation of `request.getRemoteUser()`, it gets the `Authentication` and authenticates if instance of `Authentication.Deferred`, it then gets the `UserPrincipal` name from that.

This is exactly the same as what the `CustomRequestLog` method `getAuthentication(request, true)` will do, which you should be able use with `%{d}u`.

**Comment by lachlan-roberts** [comment]
Closing and addressing these issues with the `%u` code in PR #4397.

**Comment by cstamas** [comment]
@lachlan-roberts Howdy, sorry for late response.

While I do agree somewhat with your, these were my concerns:
* I assumed (seems wrongly?) this is about a "jetty OOTB" access logger, given one of the most fav logging solution (logback access w/ Jetty support has a long outstanding issue https://jira.qos.ch/browse/LOGBACK-1052)
* Many frameworks (for example Shiro (wraps the request, hence while your code pointers stand for Jetty Request, that is clearly not the case for wrapped ones
* If this access logger is meant to be used in some special manner (ie. jetty-only), then it may need Javadoc update, or, some hint how to be used. Basically I went on (trying to use it) with wrong assumptions then.

**Comment by lachlan-roberts** [comment]
@cstamas this access logger will never see the wrapped request, it is given the jetty request directly so sees what goes over the wire and not any wrapped requests.

It is possible to configure to use Logback by using a `RequestLogWriter`, you could try using the `Slf4jRequestLogWriter` with the [Logback Implementation for Slf4j](http://logback.qos.ch/).

**Comment by cstamas** [comment]
So, then it will see only auth that jetty knows about? Then it will not work for me.

**Comment by joakime** [comment]
In short, if it's not servlet auth, then it's app specific auth.

If you can set the authentication via `Request.setAuthentication(Authentication)` on the base Jetty `Request` object you might be able to get what you need.

**Comment by joakime** [comment]
@lachlan-roberts perhaps the javadoc for `%u` should indicate that it's only for servlet auth.

**Comment by gregw** [comment]
Perhaps set the custom auth as a request attribute and then that can be logged?

**Pull request closed without merging**